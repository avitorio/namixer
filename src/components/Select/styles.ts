import styled, { css } from 'styled-components'
import { lighten } from 'polished'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
`

export const Select = styled.select`
  ${({ theme }) => css`
    color: ${theme.colors.neutral.neutral900};
    display: inline-block;
    font-family: ${theme.font.family};
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};
    padding: 1rem 3rem 1rem 2rem;
    border: 2px solid #ffffff00;
    border-radius: ${theme.border.radius.small};
    outline: none;
    appearance: none;
    background-color: ${theme.colors.neutral.neutral100};
    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEoAAABQCAYAAAC+neOMAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAAOxAAADsQBlSsOGwAAA5tJREFUeJzt2s9rHGUYwPHvM9lpKjYQFENBEIzSelHwPxBMeuhNJYtxAioehKyiNw8ik148CFLqtpG9WEI6G38h1EhPQgTvnjVCD4pVPBg8eKhJ5vHS3Wzipju7O+/7TuH5nmZnZncePpmEd4eAZVmWZVmWZVmWZVmWZVmWZVnWEEmRk9I0jU6fOfPYvsgD5PkfjST5RUTU9XCuaq6tPRhFk7NM5Lf3pqd/euv8+duD3nNXqDRNo5mzZxuq+o4gD3cPKD9C/m5jaemrEub2VvPatScj5ENE5nt2/6Oqn8Qn4vder9f/Pu69x0K1Wq147/6pDYQXjjtHlZU3ll66MOLcXmu22/NRrtcRua/vCarbTERzjcXFX/sd7gvVarXi3VNTnwo8P3gEvdBIkpXiI/vvyvrGOY30usDJu52ncFMieaYfVnR0R7q1Vds9NZUVQwKQ9EqWrRSc2XtFkQAEZjXX71prnz9y9NghqHRrqzbz2+/rAgvDjVNNrGGQOgnM7tX2/od1CGrm1q0PEF4cbaxqYY2C1NOjuxO7m5du3Jjs7OhCfZRlT6jy9njjVQOr2W7Pj4EEgIg8VdvZea3zOurZWBCRQuuqAZcIitVst+dF+XocpJ66f4IOfvVUHi/hg+8UBqtkJODApAslogNXp0NexCtW+Uig8G9nu/eO+qGsCxzkB8sFEoCgXZPeO+ozhZ0yL3Tnk51iuUICEGh1trtQy0myI+jLqrpf9gVdYa1m2ZwrJJSPl5Pk287LQ+uoRpJsRsLivYC1mmVzObLpBAldf+hE7c3ePX2XA6tZtpArGyIy4WCIsb8bOkeK41fr9fqhm+XYdVNVsUIgwYDnUVXDCoUEBZ5wVgUrJBIUfBQcGuvy+sazRPpNKCQoCAXhsKqABENAgX+sqiDBkFDgD6tKSDACFLjH0jz6vkpIMCIUuMVS1X1HP4SRkGAMKHB9Z5Xd6EgwJhTcK1jjIUEJUFB1rPGRoCQoqCpWOUhQIhRUDas8JCgZCqqCVS4SOICC0FjlI4EjKAiF5QYJHEKBbyx3SOAYCnxhuUUCD1Dg/LuhcyTwBAWusPwggUcoKBvLHxJ4hoKysPwiQQAoGBfLPxIEgoJRscIgQUAoGBYrHBIEhoKiWGGRoM+/T/tuOUm+EOE5Vf3r6DFV3Ud5/8/t7VdCIkEF7qhOF69enY7jyQWBp4EY+FlEv1xOkpuhZ7Msy7Isy7Isy7IsyxrcfzNeLxtxSlaeAAAAAElFTkSuQmCC);
    background-repeat: no-repeat, repeat;
    background-position: right 0.7em top 50%, 0px 0px;
    background-size: 0.65em, 100%;

    &:hover {
      background-color: ${lighten(0.05, theme.colors.secondary)};
    }

    ${media.greaterThan('medium')`
    padding: ${theme.spacings.xsmall} 30px ${theme.spacings.xsmall} 20px;
    `}
  `}
`
