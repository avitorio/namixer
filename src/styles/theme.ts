export default {
  grid: {
    container: '137rem',
    gutter: '3.2rem'
  },
  border: {
    radius: {
      small: '0.4rem',
      big: '0.8rem'
    }
  },
  font: {
    family:
      "Source Sans Pro, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    light: 300,
    normal: 400,
    bold: 600,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      xxxlarge: '3.6rem',
      huge: '5.2rem'
    }
  },
  colors: {
    primary: '#E65A6D',
    primaryColor: {
      primary100: '#E65A6D1A',
      primary200: '#E65A6D33',
      primary300: '#E65A6D4D',
      primary400: '#E65A6D66',
      primary500: '#E65A6D80',
      primary600: '#E65A6D99',
      primary700: '#E65A6DB3',
      primary800: '#E65A6DCC',
      primary900: '#E65A6DE6',
      primary1000: '#E65A6D'
    },
    secondary: '#C9FCC1',
    secondaryColor: {
      secondary100: '#C9FCC11A',
      secondary200: '#C9FCC133',
      secondary300: '#C9FCC14D',
      secondary400: '#C9FCC166',
      secondary500: '#C9FCC180',
      secondary600: '#C9FCC199',
      secondary700: '#C9FCC1B3',
      secondary800: '#C9FCC1CC',
      secondary900: '#C9FCC1E6',
      secondary1000: '#C9FCC1'
    },
    mainBg: '#030517',
    lightBg: '#34C1FF66',
    matisse: '#1F688E',
    white: '#FAFAFA',
    black: '#030517',
    neutral1: '#2C3A4B',
    neutral: {
      white: '#FFFFFF',
      neutral100: '#F4F6F9',
      neutral200: '#EBEEF2',
      neutral300: '#DADEE3',
      neutral400: '#A5ABB3',
      neutral500: '#858C94',
      neutral600: '#6D7580',
      neutral700: '#545D69',
      neutral800: '#394452',
      neutral900: '#2C3A4B',
      black: '#09101D'
    },
    lightGray: '#EAEAEA',
    gray: '#8F8F8F',
    darkGray: '#2E2F42',
    red: '#FF6347',
    success: '#287D3C',
    error: '#DA1414',
    warning: '#B95000'
  },
  accentColors: {
    blue: '#2E8EF7',
    green: '#44cc44',
    orange: '#FFC371',
    pink: '#FF409A',
    purple: '#C341FF'
  },
  gradients: {
    lighterBlue: `linear-gradient(180deg, rgba(54, 200, 255, 0.06) 15.1%, rgba(196, 196, 196, 0) 63.54%)`,
    blue: `linear-gradient(222deg, #36CBFF -1.24%, #2E8EF7 111.42%);`,
    lightBlue: `linear-gradient(109deg, #FFFFFF 0%, #A3DEFD 101.14%);`,
    green: `linear-gradient(213deg, #16B4EB 4.85%, #23E73C 119.17%);`,
    orange: `linear-gradient(135deg, #FF5F6D 2%, #FFC371 100%);`,
    pink: `linear-gradient(136deg, #FF409A 8.34%, #C438EF 95.26%);`,
    purple: `linear-gradient(213.66deg, #8E31FD 4.85%, #C341FF 119.17%);`
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  },
  shadow: {
    fluffy: {
      elevation5: '0 15px 60px #0000001f;'
    }
  }
} as const
