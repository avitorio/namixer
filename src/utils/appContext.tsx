import {
  Dispatch,
  SetStateAction,
  useState,
  useContext,
  createContext,
  ReactNode
} from 'react'

export interface IProviderProps {
  children?: ReactNode
}

type AppContextState = { domain: string }

type AppContextValue = {
  state: AppContextState
  setState: Dispatch<SetStateAction<AppContextState>>
}

export const AppContext = createContext<AppContextValue | undefined>(undefined)

export const AppProvider = (props: IProviderProps) => {
  const [state, setState] = useState({ domain: 'YourDomain.com' })

  return (
    <AppContext.Provider value={{ state, setState }}>
      {props.children}
    </AppContext.Provider>
  )
}

export function useAppContext() {
  const ctxValue = useContext(AppContext)
  if (ctxValue === undefined)
    throw new Error(
      'Expected an AppProvider somewhere in the react tree to set context value'
    )
  return ctxValue // now has type AppContextValue
  // or even provide domain methods for better encapsulation
}
