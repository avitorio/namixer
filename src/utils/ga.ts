// https://developers.google.com/analytics/devguides/collection/gtagjs/pages
export const pageview = (url: URL): void => {
  window.gtag('config', `${process.env.NEXT_PUBLIC_GA_TRACKING}`, {
    page_path: url
  })
}

type EventParams = { [key: string]: string | number }

type GTagEvent = {
  event: string
  params: EventParams
}

// https://developers.google.com/analytics/devguides/collection/gtagjs/events
export const event = ({ event, params }: GTagEvent): void => {
  window.gtag('event', event, {
    ...params
  })
}
