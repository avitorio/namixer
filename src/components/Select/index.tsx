import { SelectHTMLAttributes } from 'react'

import * as S from './styles'

type SelectValue = string | ReadonlyArray<string> | number

export type SelectProps = {
  onSelect?: (value?: SelectValue) => void
  children: React.ReactNode
} & SelectHTMLAttributes<HTMLSelectElement>

const Select = ({ children, onSelect, value, ...props }: SelectProps) => {
  const onChange = () => {
    !!onSelect && onSelect(value)
  }

  return (
    <S.Wrapper>
      <S.Select onChange={onChange} {...props}>
        {children}
      </S.Select>
    </S.Wrapper>
  )
}

export default Select
