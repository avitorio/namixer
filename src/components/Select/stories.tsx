import { Story, Meta } from '@storybook/react/types-6-0'
import Select, { SelectProps } from '.'

export default {
  title: 'Form/Select',
  component: Select,
  parameters: {
    layout: 'fullscreen',
    backgrounds: {
      default: 'won-dark'
    }
  },
  argTypes: {
    onCheck: { action: 'checked' }
  }
} as Meta

export const Default: Story<SelectProps> = (args) => (
  <>
    <Select id="primeiro" name="nome" {...args}>
      <option value="1">1</option>
    </Select>
  </>
)
