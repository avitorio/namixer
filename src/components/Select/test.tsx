import { render, screen } from 'utils/test-utils'
import userEvent from '@testing-library/user-event'
import theme from 'styles/theme'

import Select from '.'

describe('<Select />', () => {
  it('should render with label (white)', () => {
    const { container } = render(
      <Select id="Select" name="Select" data-testid="select">
        <option value="1">Select</option>
      </Select>
    )

    const label = screen.getByTestId('select')
    expect(label).toBeInTheDocument()
    expect(label).toHaveStyle({ color: theme.colors.neutral.neutral900 })
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should dispatch onCheck when option changes', async () => {
    const onCheck = jest.fn()
    render(
      <Select
        id="Select"
        name="Select"
        data-testid="select"
        onChange={onCheck}
        defaultValue="1"
      >
        <option value="1">A</option>
        <option value="2">B</option>
      </Select>
    )

    expect(onCheck).not.toHaveBeenCalled()
    const optionB = screen.getByRole('option', {
      name: 'B'
    }) as HTMLOptionElement

    userEvent.selectOptions(screen.getByTestId('select'), ['2'])
    expect(optionB.selected).toBe(true)
  })

  it('Should be accessible with tab', () => {
    render(
      <Select id="Select" name="Select" data-testid="select">
        <option value="1">Select</option>
      </Select>
    )

    const select = screen.getByTestId('select')

    expect(document.body).toHaveFocus()

    userEvent.tab()

    expect(select).toHaveFocus()
  })
})
