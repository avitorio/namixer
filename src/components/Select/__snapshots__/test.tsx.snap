// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render with label (white) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  color: #2C3A4B;
  display: inline-block;
  font-family: Source Sans Pro,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  font-weight: 600;
  padding: 1rem 3rem 1rem 2rem;
  border: 2px solid #ffffff00;
  border-radius: 0.4rem;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #F4F6F9;
  background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEoAAABQCAYAAAC+neOMAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAAOxAAADsQBlSsOGwAAA5tJREFUeJzt2s9rHGUYwPHvM9lpKjYQFENBEIzSelHwPxBMeuhNJYtxAioehKyiNw8ik148CFLqtpG9WEI6G38h1EhPQgTvnjVCD4pVPBg8eKhJ5vHS3Wzipju7O+/7TuH5nmZnZncePpmEd4eAZVmWZVmWZVmWZVmWZVmWZVnWEEmRk9I0jU6fOfPYvsgD5PkfjST5RUTU9XCuaq6tPRhFk7NM5Lf3pqd/euv8+duD3nNXqDRNo5mzZxuq+o4gD3cPKD9C/m5jaemrEub2VvPatScj5ENE5nt2/6Oqn8Qn4vder9f/Pu69x0K1Wq147/6pDYQXjjtHlZU3ll66MOLcXmu22/NRrtcRua/vCarbTERzjcXFX/sd7gvVarXi3VNTnwo8P3gEvdBIkpXiI/vvyvrGOY30usDJu52ncFMieaYfVnR0R7q1Vds9NZUVQwKQ9EqWrRSc2XtFkQAEZjXX71prnz9y9NghqHRrqzbz2+/rAgvDjVNNrGGQOgnM7tX2/od1CGrm1q0PEF4cbaxqYY2C1NOjuxO7m5du3Jjs7OhCfZRlT6jy9njjVQOr2W7Pj4EEgIg8VdvZea3zOurZWBCRQuuqAZcIitVst+dF+XocpJ66f4IOfvVUHi/hg+8UBqtkJODApAslogNXp0NexCtW+Uig8G9nu/eO+qGsCxzkB8sFEoCgXZPeO+ozhZ0yL3Tnk51iuUICEGh1trtQy0myI+jLqrpf9gVdYa1m2ZwrJJSPl5Pk287LQ+uoRpJsRsLivYC1mmVzObLpBAldf+hE7c3ePX2XA6tZtpArGyIy4WCIsb8bOkeK41fr9fqhm+XYdVNVsUIgwYDnUVXDCoUEBZ5wVgUrJBIUfBQcGuvy+sazRPpNKCQoCAXhsKqABENAgX+sqiDBkFDgD6tKSDACFLjH0jz6vkpIMCIUuMVS1X1HP4SRkGAMKHB9Z5Xd6EgwJhTcK1jjIUEJUFB1rPGRoCQoqCpWOUhQIhRUDas8JCgZCqqCVS4SOICC0FjlI4EjKAiF5QYJHEKBbyx3SOAYCnxhuUUCD1Dg/LuhcyTwBAWusPwggUcoKBvLHxJ4hoKysPwiQQAoGBfLPxIEgoJRscIgQUAoGBYrHBIEhoKiWGGRoM+/T/tuOUm+EOE5Vf3r6DFV3Ud5/8/t7VdCIkEF7qhOF69enY7jyQWBp4EY+FlEv1xOkpuhZ7Msy7Isy7Isy7IsyxrcfzNeLxtxSlaeAAAAAElFTkSuQmCC);
  background-repeat: no-repeat,repeat;
  background-position: right 0.7em top 50%,0px 0px;
  background-size: 0.65em,100%;
}

.c1:hover {
  background-color: #defdd9;
}

@media (min-width:768px) {
  .c1 {
    padding: 1.6rem 30px 1.6rem 20px;
  }
}

<div
  class="c0"
>
  <select
    class="c1"
    data-testid="select"
    id="Select"
    name="Select"
  >
    <option
      value="1"
    >
      Select
    </option>
  </select>
</div>
`
